### PR TITLE
python3.pkgs.complycube: init at 1.1.6

### PR DESCRIPTION
**Pull request by DerDennisOP**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes (or backporting 23.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by DerDennisOP** [comment]
updated to version 1.1.6

File: pkgs/development/python-modules/complycube/default.nix [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+  src = fetchPypi rec {
+    inherit version format;
+    pname = "complycube";
+    dist = python;
+    python = "py3";
+    hash = "sha256-vvhRftkB4GJRS3shpmAPR4hKX4T6vHR6oNl6OQSSXYE=";
+  };
+
+  nativeCheckInputs = [ pytest ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/complycube/default.nix:24]:
```suggestion
  nativeCheckInputs = [ pytestCheckHook ];
```
this allow us to have a stable interface to disable tests

File: pkgs/development/python-modules/complycube/default.nix [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+, pytestCheckHook
+, python
+, requests
+, setuptools
+}:
+
+buildPythonPackage rec {
+  pname = "complycube";
+  version = "1.1.6";
+  format = "wheel";
```
Comment (natsukium) [inline pkgs/development/python-modules/complycube/default.nix:14]:
```suggestion
  pyproject = true;
```
Can we use sdist?

File: pkgs/development/python-modules/complycube/default.nix [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  };
+
+  nativeCheckInputs = [
+    pytestCheckHook
+  ];
+
+  propagatedBuildInputs = [
+    pyhumps
+    requests
+    setuptools
```
Comment (natsukium) [inline pkgs/development/python-modules/complycube/default.nix:31]:
```suggestion
```
setuptools should go to nativeBuildInputs.

File: pkgs/development/python-modules/complycube/default.nix [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  ];
+
+  pyImportsCheck = [
```
Comment (natsukium) [inline pkgs/development/python-modules/complycube/default.nix:34]:
```suggestion
  pythonImportsCheck = [
```

**Review by natsukium — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/complycube/default.nix (hunk 2 above)
Comment (DerDennisOP) [reply pkgs/development/python-modules/complycube/default.nix:14]: I'm not sure, they are just releasing the wheel

Comment (natsukium) [reply pkgs/development/python-modules/complycube/default.nix:12]: No, they also provide source code.
https://pypi.org/project/complycube/#files

Comment (DerDennisOP) [reply pkgs/development/python-modules/complycube/default.nix]: here you go! :)

**Review by SuperSandro2000 — APPROVED** [review]

File: pkgs/development/python-modules/complycube/default.nix [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+  version = "1.1.6";
+  pyproject = true;
+
+  src = fetchPypi rec {
+    inherit version;
+    pname = "complycube";
+    hash = "sha256-hetcn5RX582CRVmtG5dAvr+NXD+7NKJjaqgOo8LlpqM=";
+  };
+
+  nativeCheckInputs = [
```
Comment (natsukium) [inline pkgs/development/python-modules/complycube/default.nix:20]:
```suggestion
  nativeBuildInputs = [
```
Besides this, LGTM

**Comment by natsukium** [comment]
@ofborg build python311Packages.complycube

**Review by natsukium — APPROVED** [review]

**Pull request merged**